src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -3692,6 +3694,16 @@ void EmitSingleCharRepeater(RegexNode node, bool emitLengthCheck = true)
 
                     sliceStaticPos += iterations;
                 }
+
+                static bool CanUseIndexOfAnyValues(RegexNode node)
```

Comment:
Can we put this onto RegexNode or somewhere else shared so that we don't have to duplicate it between the source generator and RegexCompiler?